plugins/org.eclipse.elk.alg.topdownpacking/docs/topdownlayout.md

```diff
@@ -0,0 +1,81 @@
+
+#### Child Area Width and Child Area Height
+`org.eclipse.elk.topdown.childAreaWidth: double`, `org.eclipse.elk.topdown.childAreaHeight: double`
+These properties are used internally to determine the size occupied by a layout. The properties are there so that algorithms that already compute these values can store them directly for later use. Otherwise they are computed by the recursive layout engine.
+
+#### Topdown Scale Cap
+`org.eclipse.elk.topdown.scaleCap: double`
+This value sets an upper limit on the resulting topdown scale factor that is possible. The default value is 1, which means that child nodes will never be scaled up. This means that they may not fully use the available space, but they also do not appear larger (relative with respect to their texts, line widths etc.) than their parents as this seems like generally undesirable behaviour. This property allows some flexibility to configure this within a graph.
+
+#### Topdown Size Approximator
+`org.eclipse.elk.topdown.sizeApproximator: TopdownSizeApproximator`
+A size approximator can only be set on hierarchical nodes and will use a given ELK node to determine a suitable size for the node before computing its layout. The `COUNT_CHILDREN` size approximator uses `sqrt(number_of_children)` as proportional factor for the node sizes. Size approximators can be made arbitrarily complex.
+
+## Requirements
+For the result layouts to be rendered correctly the viewer must support the topdown scale factor property as demonstrated in [klighd-vscode/mka/topdown](https://github.com/kieler/klighd-vscode/tree/mka/topdown). When using KlighD to create topdown diagrams, hierarchical nodes require an explicit child area to be defined so that the viewer will know what part of the rendering the topdown scale factor needs to be applied to.
+
+## Example Diagrams
+The below images show some ways that topdown layout can be used.
+
+### Layered, Force and Stress used in hierarchical nodes
+![topdown_example_combined1](/images/topdown_example_combined1.png)
+
+### Hierarchical and parallel nodes alternating
+SCCharts provide a practical use case for alternating hierarchical and parallel nodes because this structure is already inherent in the diagrams through the alternation of regions and states.
+![hierarchical_parallel_example](/images/hierarchical_parallel_example.png)
+
+### Large SCChart
+
+![topdown_example5](/images/controller.png)
+
+### Large mixed example
+All these features can be mixed quite flexibly, but it can become quite complex. Here is an example showing a mixed composition of node types and algorithms.
+![mixed_example](/images/mixed_example.png)
```

Comment:
The text in the image looks wrong and text in leaf nodes seems to be too high. Does this also happen in VS Code or online in the image?

Reply:
the labels are just default labels and not properly placed with sensible styles. It looks the same in vscode for this example. I could make a nicer example, but since the placement of the texts is not really part of what topdown layout does I hadn't done that previously